### PR TITLE
Replace pipe with RecordRTC

### LINKED ISSUE #307: Replace existing video recorder
# Summary

Currently, EFP uses [Pipe](www.addpipe.com) to record participants.  The expenses for the Pipe have risen, and with their non-response to security reviews, we've decided to replace Pipe with our own implementation.  This change will heavily lean on the [RecordRTC](https://recordrtc.org/) library.  

# Implementation 
Create a class that implements the `record`, `stop`, `getTime`, `install` and `destroy` methods from `app/services/video-recorder.js`.  A few of these methods might not be needed once we get into the RecordRTC API.  Additionally, the hook design pattern from Pipe in addition to the promise pattern is a bit murky to wade through.  If possible, we should do either one and not both.  

When video is recorded, the file should be downloaded to client.  This will be a placeholder until we get to the upload to amazon issue completed (Issue number to come). 

Generally, we should change as little as possible with this API to save time on implementing.  This Feature will require quite a bit of QC and we should spend our time there.  

# AC

- [x] Implement methods
- [x] Implement Hooks

### PR DESCRIPTION
**Pull request by becky-gilbert**
This PR replaces the existing Pipe recording system with the non-Pipe/RecordRTC version.

Fixes #307

### CODE REVIEW

**Review by bleonar5 — APPROVED** [review]

File: app/components/exp-lookit-stop-recording/component.js [hunk 1]
```diff
@@ -154,7 +159,7 @@ export default ExpFrameBaseComponent.extend(ExpandAssets, {
                 _this.send('setTimeEvent', 'warningUploadTimeoutError');
                 _this.send('next');
             }
-        }, 5000));
+        }, 600000));
```
Comment (okaycj) [inline app/components/exp-lookit-stop-recording/component.js:162]: Did we intend to leave this timeout this high?

File: app/components/exp-video-config-quality/component.js [hunk 2]
```diff
@@ -438,7 +460,61 @@ export default ExpFrameBaseComponent.extend(VideoRecord, {
...
+            this.setupRecorderAndUpdateDisplay();
+        },
+
+        playvideo() {
+            document.getElementById('recordingIndicator').classList.remove('text-danger','Blink');
+            const vid = document.querySelector('video');
+            vid.src = null;
+            vid.srcObject = null;
+            vid.muted = false;
+            vid.volume = 1;
```
Comment (okaycj) [inline app/components/exp-video-config-quality/component.js:473]: I am seeing this in two places.  Should this be de-duplicated?

File: app/mixins/video-record.js [hunk 3]
```diff
@@ -142,14 +153,22 @@ export default Ember.Mixin.create({
      * @default 1
      * @private
      */
-    autosave: 1,
+    //autosave: 1,
```
Comment (okaycj) [inline app/mixins/video-record.js:157]: Maybe this was accidentally left in.

File: app/mixins/video-record.js [hunk 4]
```diff
@@ -142,14 +153,22 @@ export default Ember.Mixin.create({
...
 
     /**
      * Whether to do audio-only (vs also video) recording. Can be overridden by consuming frame.
      * @property {Number} audioOnly
      * @default 0
      */
-    audioOnly: 0,
+    //audioOnly: 0,
+
+    /**
```
Comment (okaycj) [inline app/mixins/video-record.js:165]: Another commented out property.

**Review by okaycj — APPROVED** [review]
Whoah.  That's a lot of work.  It all looks good from here.

File: app/mixins/video-record.js (hunk 4 above)
Comment (becky-gilbert) [reply app/mixins/video-record.js]: Fixed, thanks!

File: app/mixins/video-record.js (hunk 3 above)
Comment (becky-gilbert) [reply app/mixins/video-record.js]: Fixed, thanks!

File: app/components/exp-video-config-quality/component.js (hunk 2 above)
Comment (becky-gilbert) [reply app/components/exp-video-config-quality/component.js]: Great catch - thanks!! I've refactored this to remove the duplication.

File: app/components/exp-lookit-stop-recording/component.js (hunk 1 above)
Comment (becky-gilbert) [reply app/components/exp-lookit-stop-recording/component.js]: Another great catch, thanks for flagging this. I've reduced this from 10 to 5 minutes (and updated the comment about it). I'm not sure what a reasonable duration is for completion of the first part of a multi-part upload on slow connections. I'd like to err on the side of not losing data, and decrease this duration if we get complaints about it. I will also make a note to test this myself on a throttled connection.

**Pull request merged**